### PR TITLE
Fix Dispose calls and only allow static OneTime- methods

### PR DESCRIPTION
**Pull request by gleb-osokin**
This one takes care of the comments on the original PR:
1) Disallow instance OneTimeSetUp and OneTimeTearDown methods on fixtures marked with LifeCycle.InstancePerTestCase
2) Fix dispose calls for InstancePerTestCase fixtures.

I have also adjusted the tests to cover the case from @rprouse (i.e., dispose calls in specific order).

### CODE REVIEW

File: src/NUnitFramework/framework/Internal/Execution/IMethodValidator.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using System.Reflection;
```
Comment (rprouse) [inline src/NUnitFramework/framework/Internal/Execution/IMethodValidator.cs:24]: Can you please add the copyright header to this file?

**Review by rprouse — COMMENTED** [review]
I'll do a more thorough review once this is incoming into the NUnit project. I'll leave it up to @avilv on how he wants to do that, by accepting this PR or starting a new one in NUnit.

File: src/NUnitFramework/framework/Internal/Execution/IMethodValidator.cs (hunk 1 above)
Comment (gleb-osokin) [reply src/NUnitFramework/framework/Internal/Execution/IMethodValidator.cs:24]: Thanks, added

**Comment by rprouse** [comment]
I'm just sitting down to test these changes, thanks for merging them @avilv, that makes it so much easier. I'm stoked to get this into NUnit.

**Pull request merged**